jenkins/pipelines/cd/nightly/build-darwin-amd64-4.0.groovy

```diff
@@ -499,16 +501,75 @@ try {
     }
 
     currentBuild.result = "SUCCESS"
-} catch (Exception e) {
-    currentBuild.result = "FAILURE"
-    slackcolor = 'danger'
-    echo "${e}"
+} catch (org.jenkinsci.plugins.workflow.steps.FlowInterruptedException e) {
+    println e
+    // this ambiguous condition means a user probably aborted
+    currentBuild.result = "ABORTED"
+} catch (hudson.AbortException e) {
+    println e
+    // this ambiguous condition means during a shell step, user probably aborted
+    if (e.getMessage().contains('script returned exit code 143')) {
+        currentBuild.result = "ABORTED"
+    } else {
+        currentBuild.result = "FAILURE"
+    }
+} catch (InterruptedException e) {
+    println e
+    currentBuild.result = "ABORTED"
+}
+catch (Exception e) {
+    if (e.getMessage().equals("hasBeenTested")) {
+        currentBuild.result = "SUCCESS"
+    } else {
+        currentBuild.result = "FAILURE"
+        slackcolor = 'danger'
+        echo "${e}"
+    }
 }
 
 stage('Summary') {
+    def duration = ((System.currentTimeMillis() - taskStartTimeInMillis) / 1000 / 60).setScale(2, BigDecimal.ROUND_HALF_UP)
     echo "Send slack here ..."
     //slackSend channel: "", color: "${slackcolor}", teamDomain: 'pingcap', tokenCredentialId: 'slack-pingcap-token', message: "${slackmsg}"
     // if (currentBuild.result != "SUCCESS") {
     //     slackSend channel: '#jenkins-ci-build-critical', color: 'danger', teamDomain: 'pingcap', tokenCredentialId: 'slack-pingcap-token', message: "${slackmsg}"
     // }
-}
\ No newline at end of file
+    
+    // send a Lark message about result, now it only send tiflash compilation result.
+    stage("sendLarkMessage") {
+        print "currentBuild.result=${currentBuild.result}"
+        if (currentBuild.result == "ABORTED") {
+            tiflash_result = "ABORTED"
+        }
+        def result_mark = "❌"
+        if (tiflash_result == "ABORTED" || tiflash_result == "NOT TRIGGERED") {
+            result_mark = "🟡"
+        } 
+        if (tiflash_result == "SUCCESS") {
+            result_mark = "✅"
+        }
+
+        def feishumsg = "tiflash_darwin_amd64_build_daily\\n" +
+                "Build Number: ${BUILD_NUMBER}\\n" +
+                "Result: ${tiflash_result} ${result_mark}\\n" +
+                "RELEASE_TAG: ${RELEASE_TAG}\\n" +
```

Comment:
What is a "release tag"?

Reply:
Seems like a git tag?

Reply:
it can be used to build another branch's nightly, for example, 5.0-nightly, 5.1-nightly, like https://cd.pingcap.net/blue/organizations/jenkins/build-darwin-amd64-4.0/detail/build-darwin-amd64-4.0/84/pipeline

Reply:
This is a variable that was passed in when the pipeline is started. Normally, we pass in `nightly`. In addition, it can also be `v5.1.0-nightly`, `v5.0.0-nightly`.

Reply:
I saw it could also be a specific git tag: 5.0.1.

Reply:
seems have some other issue:

https://cd.pingcap.net/blue/organizations/jenkins/build-linux-arm64-4.0/detail/build-linux-arm64-4.0/61/pipeline
![image](https://user-images.githubusercontent.com/30543181/123939723-06c38f00-d9cb-11eb-9d08-0d8a062d85c4.png)

Reply:
v5.0.1 same
![image](https://user-images.githubusercontent.com/30543181/123940011-55712900-d9cb-11eb-80cb-6b917dd576cb.png)

Reply:
oh, it's blocked by lighting.